CMakeLists.txt

```diff
@@ -27,6 +27,11 @@ endif()
 # set the minimum required version across the board
 cmake_minimum_required(VERSION 3.1.0)
 
+# for link time optimization, remove if cmake version is >= 3.9
+if(POLICY CMP0069)
+    cmake_policy(SET CMP0069 NEW)
+endif()
+
```

Comment:
Why are we messing with the policy? IIRC, the history of the option was:

* It didn't exist. (This was pre-CMake-3.0, and I'm pretty sure we use CMake 3.0 stuff to link with RecastNavigation, so this reality isn't relevant to us.)
* It was added. Turning it on would enable LTO when compatible build tools were used, and silently fail (or maybe just log that it was incompatible) if incompatible tools were used.
* CMake 3.9 came and added policy CMP0069 and `check_ipo_supported`. If you turned the option on for an unsupported set of tools, it was now an error, so if you wanted to turn it into a warning or give a custom error message, you needed to do the new check.

To me, this means that the most sensible option (and what I *think* the snippet of CMake's own `CMakeLists.txt` does) is:
* Check the option is set, if it is, keep going, otherwise, skip over all of this.
* If CMake ≥ 3.9 is used and the policy is set to the new value (checking the policy on its own is fine as that implies the version is sufficient), then checking the new check is necessary, so do that, otherwise skip to setting the setting.
* If the check fails, yell at the user and don't set the option.
* Set the option.

Reply:
> Why are we messing with the policy?
>     * It didn't exist. (This was pre-CMake-3.0 ...

Policy CMP0069 was introduced in cmake 3.9
https://cmake.org/cmake/help/v3.9/policy/CMP0069.html

> It was added. Turning it on would enable LTO ...
If i have understood correctly that is not what it does
`This policy takes effect whenever the IPO property is enabled. The OLD behavior for this policy is to add IPO flags only for Intel compiler on Linux. The NEW behavior for this policy is to add IPO flags for the current compiler or produce an error if CMake does not know the flags.`

So you want me to check against the cmake version 3.9 instead of checking if the policy that was added in this verson exists ?

Maybe the confusion is this:
The example posted in the ticket uses:
cmake_minimum_required(VERSION 3.1...3.15 FATAL_ERROR)
we use:
cmake_minimum_required(VERSION 3.1.0)
wich forces the policy to be unset.
https://cmake.org/cmake/help/v3.1/command/cmake_policy.html#setting-policies-by-cmake-version

Reply:
I'm pretty sure we also set the minimum version to other values in other places, so now you've reminded me that setting the minimum version also sets its policies, I'm not even sure this will stick throughout the whole CMake process.

My preference would be to make the place where we actually set it able to work no matter what the policy is set to. If it turns out that the old policy also uses CMake 3.8's restricted list of tools rather than just mandating the check, then we should set the new policy right before we do the thing that requires it.